modules/client/core/services/socket.io.client.service.ts

```diff
@@ -1,15 +1,35 @@
 'use strict';
 import angular from '../../../../node_modules/angular'
+import * as React from 'react'
+import * as ReactDOM from 'react-dom'
 import {coreModule} from "../core.client.module";
 import {authenticationService} from "../../users/services/authentication.client.service";
 
 export const currentSocket = {
   socket: null,
 }
 
+const eventToCallback: {[event: string]: (arg: any) => void} = {}
+
+export const useAddSocketListener = ((event: string, callback: (x: any) => void) => {
+  eventToCallback[event] = callback
+
+  const runEventCallback = (arg) => {
+    // We need to force render React because callbacks might update React state, and we need subsequent callbacks to
+    // read the latest React state (that may have been updated from previous callbacks or elsewhere).
+    ReactDOM.flushSync(() => {
+      eventToCallback[event](arg)  // this works
+    })
+  }
+  React.useEffect(() => {
+    currentSocket.socket.addEventListener(event, runEventCallback)
+    return (() => currentSocket.socket.removeEventListener(event, runEventCallback))
+  }, [])
+})
+
 export const connectSocket = (user) => {
   // Connect only when authenticated
-  if (user) {
+  if (user && !currentSocket.socket) {
```

Comment:
this is needed to ensure we only have one socket across the react and angular code. otherwise the server thinks we are 2 clients